### PR TITLE
Readlen functions testing benjamin

### PR DESCRIPTION
**Pull request by BGSpiegl**
changes for computeGCBias_readlen() and correctGCBias_readlen() functions:
 - solved some issues related to variable scope
 - removed unused variables
 - added missing import concerning BED graph processing
 - reformatted code for closer PEP adherence

### CODE REVIEW

**Comment by sroener** [comment]
Hi @BGSpiegl,

thank you for all the changes regarding the PEP adherence!

Just to make sure that I understand the intention of defining the debug variable in the tabulateGCcontent_worker( ) function as global: The purpose is to support proper logging when using the Tester class, right?

**Comment by BGSpiegl** [comment]
In short: yes, also that (but one would have to ask the deepTools devs for theirs). Using 'debug' as global inside the Tester initialization and setting the value to 0 ensures that the output will not be affected by the logging level set at runtime before using any Tester object. Not doing this could make tests fail (some function call might set the "logging level" differently).

A longer comment from my side: currently, the logging is not optimal in this script. Either one uses the logging module and defines the logging level once (e.g. "Logger.setLevel()"), or one can implement it themselves. In computeGCBias_readlen.py, the logging style is actually mixed because the global logging level is disregarded by using a bool called 'debug' which might be true leading to debugging output will be logged, or vice versa.
The logging module offers a mechanism which allows one to use logging.debug without asking for a specific level - it is just not logged if the global logging level higher than debug (e.g. ERROR). See [python3 docs](https://docs.python.org/3/library/logging.html#levels) for clarification.
The "debug" state is set once for the module and then used by all functions and objects. One could also create a logger and set the level and remove the  'debug' variable. Or one could inject the debugging level / the debug variable from the main function into the functions it is calling. The logging level must be set to a fixed value when using the Tester class.
I would prefer using only the logging module way. But when using a Tester object, it might be easier to handle having a single variable that silences debugging output.

The problem concerning the debug variable and why I even touched it (I am now referring to file computeGCBias_readlen.py,  line 280 in tabulateGCcontent_wrapper()):
The (global) variable 'debug' is used without being defined in line 396 in the function. Looking at the original deepTools code, the variable was defined at module level as 0.
As a result, upon reaching line 396 of the code, the python interpreter searches outer scopes until it either finds an instance called "debug" and use it, or it will give a "variable not defined" exception at runtime, which was the case for me when using the script. Therefore, I set 'debug' explicitly to global in the function. From a program flow point of view, debug should have a default value defined at module level (top of file; I would prefer a falsy value).
It could be offered to the user as a command-line flag (e.g. --debug) which is parsed/set directly in main after receiving command-line arguments. Previously, the use of 'debug' was somewhat obscure (not defined at module level) and it could not be changed by the user, only by changing the code itself. (how useful is a variable that is always zero/falsy?)

I pushed an update implementing debug as a cmd-line flag. This was missing in the current 2 commits.

Following this "make it explicit" strategy, I defined some global variables at module level, providing default values, which increases the legibility of the script and avoids raising runtime exceptions.
In general, when using globals, it should be clear how the variable is initialized and it should not be set for the module level inside some function, especially if it is set in a function outside of main. Currently, the definition inside main is also somewhat unclean - one could remove e.g. line 516 in correctGCBias_readlen.py.

Long story short: merging including the next commit would make more sense.

**Comment by sroener** [comment]
Again, thank you for the detailed answer. 

I'm completly with you regarding the uniform logging via the logging module. I switched to the logging module from the original approach of printing everything to stdout, but havn't found the time to properly rework the whole logging process. Same goes for a lot of the original code. I will spend some time on refactoring, after everything works as intended (no more weird outliers). 

I was also thinking about ditching the tester classes and move to pytest. Do you have any thoughts on this?

**Comment by BGSpiegl** [comment]
I like pytest. A solution using it instead of a special class would be more clear imho and might even reduce the code length.
